osf/management/commands/transfer_quickfiles_to_projects.py

```diff
@@ -31,145 +30,127 @@
+            target_content_type=ContentType.objects.get_for_model(AbstractNode)
         )
-
-        quick_files_nodes = AbstractNode.objects.filter(
-            id__in=quick_files_node_with_files_ids,
-        ).order_by('pk')
-
-        Guid.objects.filter(
-            id__in=quick_files_nodes.values_list('guids__id', flat=True)
-        ).delete()
-
-        # generate unique guids prior to record creation to avoid collisions.
-        guids = set([])
-        while len(guids) < quick_files_node_with_files_ids.count():
-            guids.add(generate_guid())
-        guids = list(guids)
-
-        guids = [
-            Guid(
-                _id=_id,
-                object_id=node,
-                content_type=ContentType.objects.get_for_model(Node)
-            ) for _id, node in zip(guids, quick_files_node_with_files_ids)
-        ]
-        Guid.objects.bulk_create(guids)
-
-        node_logs = [
-            NodeLog(
-                node=node,
-                user=node.creator,
-                original_node=node,
-                params={'node': node._id},
-                action=NodeLog.MIGRATED_QUICK_FILES
-            ) for node in quick_files_nodes
-        ]
-        NodeLog.objects.bulk_create(node_logs)
-
-        queued_mail = [
-            QueuedMail(
-                user=node.creator,
-                to_addr=node.creator.email,
-                send_at=QUICKFILES_DATE,
-                email_type=mails.QUICKFILES_MIGRATED.tpl_prefix,
-                data=dict(
-                    osf_support_email=settings.OSF_SUPPORT_EMAIL,
-                    can_change_preferences=False,
-                    quickfiles_link=node.absolute_url
-                )
-            ) for node in quick_files_nodes
-        ]
-        QueuedMail.objects.bulk_create(queued_mail)
-
-        for i, node in enumerate(quick_files_nodes):
-            # update guid in logs
-            for log in node.logs.all():
-                log.params['node'] = node._id
-                log.save()
-
-            if i and not i % page_size:
-                logger.info(f'{i} quickfiles were projectified at {datetime.datetime.now()}')
-
-        quick_files_nodes.update(description=QUICKFILES_DESC, type='osf.node')
-
-        logger.info(f'all quickfiles with files were projectified.')
-
-
-def reverse_remove_quickfiles(dry_run=False, page_size=1000):
-    with transaction.atomic():
-        quickfiles_nodes_with_files = Node.objects.filter(
-            logs__action=NodeLog.MIGRATED_QUICK_FILES
+    )
+    quick_files_nodes = QuickFilesNode.objects.annotate(has_files=quick_file_annotation).filter(has_files=True)
+    target_count = quick_files_nodes.count()
+    logger.info(f'Acquired {target_count} targets')
+
+    _ = Guid.objects.filter(
+        id__in=quick_files_nodes.values_list('guids__id', flat=True)
+    ).delete()
+    logger.info(f'Deleted guids: {_}')
+
+    # generate unique guids prior to record creation to avoid collisions.
```

Comment:
Because generate_guid checks againest the DB, but it doesn't check against IDs it just generated itself. So we must use a set because `generate_guid` will have non-unique entries in that list. When generating 13,000 Ids their is certain to be a collision eventually even in a batch of 750 there's a 1% chance there's a collision before to even check per-exsisting guids. 

Like just imagine the biggest coincidence where `generate_guid`  makes two guids exactly the same two times in a row, how do we weed out the duplicates before they enter the DB? I am just using a set here to enforce uniqueness,